spec.html

```diff
@@ -25475,10 +25575,155 @@ <h1>
+              1. For each Cyclic Module Record _m_ of _module_.[[AsyncParentModules]], do
+                1. If _execList_ does not contain _m_ and _m_.[[CycleRoot]].[[EvaluationError]] is ~empty~, then
+                  1. Assert: _m_.[[Status]] is ~evaluating-async~.
+                  1. Assert: _m_.[[EvaluationError]] is ~empty~.
+                  1. Assert: _m_.[[AsyncEvaluation]] is *true*.
+                  1. Assert: _m_.[[PendingAsyncDependencies]] &gt; 0.
+                  1. Set _m_.[[PendingAsyncDependencies]] to _m_.[[PendingAsyncDependencies]] - 1.
+                  1. If _m_.[[PendingAsyncDependencies]] = 0, then
+                    1. Append _m_ to _execList_.
+                    1. If _m_.[[Async]] is *false*, perform ! GatherAsyncParentCompletions(_m_, _execList_).
+              1. Return *undefined*.
+            </emu-alg>
+            <emu-note>
+              <p>When an async execution for a root _module_ is fulfilled, this function determines the list of modules which are able to synchronously execute together on this completion, populating them in _execList_.</p>
+            </emu-note>
+          </emu-clause>
+
+          <emu-clause id="sec-async-module-execution-fulfilled" type="abstract operation">
+            <h1>
+              AsyncModuleExecutionFulfilled (
+                _module_: a Cyclic Module Record,
+              )
+            </h1>
+            <dl class="header">
+            </dl>
+            <emu-alg>
+              1. If _module_.[[Status]] is ~evaluated~, then
+                1. Assert: _module_.[[EvaluationError]] is not ~empty~.
+                1. Return *undefined*.
+              1. Assert: _module_.[[Status]] is ~evaluating-async~.
+              1. Assert: _module_.[[AsyncEvaluation]] is *true*.
+              1. Assert: _module_.[[EvaluationError]] is ~empty~.
+              1. Set _module_.[[AsyncEvaluation]] to *false*.
+              1. Set _module_.[[Status]] to ~evaluated~.
+              1. If _module_.[[TopLevelCapability]] is not ~empty~, then
+                1. Assert: _module_.[[CycleRoot]] is _module_.
+                1. Perform ! Call(_module_.[[TopLevelCapability]].[[Resolve]], *undefined*, &laquo; *undefined* &raquo;).
+              1. Let _execList_ be a new empty List.
+              1. Perform ! GatherAsyncParentCompletions(_module_, _execList_).
+              1. Let _sortedExecList_ be a List whose elements are the elements of _execList_, in the order in which they had their [[AsyncEvaluation]] fields set to *true* in InnerModuleEvaluation.
+              1. Assert: All elements of _sortedExecList_ have their [[AsyncEvaluation]] field set to *true*, [[PendingAsyncDependencies]] field set to 0, and [[EvaluationError]] field set to *undefined*.
+              1. For each Cyclic Module Record _m_ of _sortedExecList_, do
+                1. If _m_.[[Status]] is ~evaluated~, then
+                  1. Assert: _m_.[[EvaluationError]] is not ~empty~.
```

Comment:
I think this is about AsyncModuleExecutionRejected being synchronously recursive, where all the async parents chains have their errors recorded. So if we hit an async parent in the fulfilled path that's already evaluated, it's because there was a transitive rejection that happened during some iteration. Fulfillment OTOH doesn't work like that and needs to wait for all the async dependencies to fulfill.

Reply:
I don't have a good note to suggest, so I'm going to leave this alone.